pkgs/development/python-modules/openai-triton/default.nix

```diff
@@ -57,6 +59,16 @@ buildPythonPackage rec {
     #   url = "https://github.com/openai/triton/commit/fc7c0b0e437a191e421faa61494b2ff4870850f1.patch";
     #   hash = "sha256-f0shIqHJkVvuil2Yku7vuqWFn7VCRKFSFjYRlwx25ig=";
     # })
+  ] ++ lib.optionals (!cudaSupport) [
+    # Manually add the ptxas version md5 sum
+    # This allows us to ensure compatibility with other derivations
+    # that may have `openai-cuda` or `cudaPackages` as an input
+    # Check this every update of (this package's version of) cudaPackages and this derivation!
+    # (`cudaPackages_12_0.cuda_nvcc`) `ptxas --version | md5sum`
+    (substituteAll {
+      src = ./0000-strip-and-replace-ptxas.patch;
+      ptxas_version = "0afa0fd697019085a7525fcab2503fcf";
+    })
```

Comment:
Could we please try the [try-except](https://github.com/openai/triton/compare/main...SomeoneSerge:openai-triton:feat/cuda-optional#diff-ffa9512af1b322c34a998219b5601a203c6a301217ddf04e0adb6f791f72cf74R177-R182) thingy linked before and get rid of the hash?

Reply:
EDIT: actually, in the triton version used in this PR you maybe don't even have to patch that bit, because they already `catch Exception`?

Reply:
Yes, that's why I just kept the current patch. (and avoided a mass rebuild)

Reply:
The reason I keep the hash is for programs that expect it.
This like the comment says, ensures compatibility, since this isn't an official fix.
It's just a failsafe, any derivation we're building that requires openai-triton that doesn't have cuda support, shouldn't be expecting cuda and relying on that hash, but it's there just in case.

Reply:
I don't think there are any programs that expect anything about this "key"? As far as I can see, the whole thing ultimately ends up fed into a hash function and used to version the jit cache: https://github.com/openai/triton/blob/2323adb38743384c557a85744d1256892754de28/python/triton/compiler/make_launcher.py#L20